<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PresentCreator</class>
 <widget class="QDialog" name="PresentCreator">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>476</width>
    <height>371</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Present Settings</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="2" rowspan="2">
    <widget class="Line" name="LineV_1">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
    </widget>
   </item>
   <item row="1" column="0" colspan="2">
    <widget class="QLineEdit" name="LE_PageSearch">
     <property name="maxLength">
      <number>40</number>
     </property>
     <property name="placeholderText">
      <string>Search for Item ID / Item Name In Page</string>
     </property>
     <property name="clearButtonEnabled">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="0" column="3">
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Save</set>
     </property>
    </widget>
   </item>
   <item row="0" column="0">
    <widget class="QLineEdit" name="LE_ItemID">
     <property name="toolTip">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;This Item ID corresponds to which item the player will recieve in the present.&lt;/p&gt;&lt;p&gt;These Item IDs are from before the Welcome Amiibo Update. The payload uses the old Item ID and the game 'translates' the old Item ID to the respective Welcome Amiibo Item ID by just using an array of the new Item IDs and the old Item ID as the index. &lt;/p&gt;&lt;p&gt;This means the devs were either 1) Lazy, 2) Lost the source code of their official present creation tool and had to create a workaround, 3) Wanted to keep old payloads compatible with Welcome Amiibo. (Likely 3 but it'd be pretty Nintendo if it was 2)&lt;/p&gt;&lt;p&gt;This therefore means that &lt;span style=&quot; font-weight:600;&quot;&gt;NO&lt;/span&gt; Items introduced in the Welcome Amiibo update are compatible with Presents :(&lt;/p&gt;&lt;p&gt;The built-in browser below shows a list of Item IDs that are compatible (Note: &lt;span style=&quot; text-decoration: underline;&quot;&gt;This requires an internet connection&lt;/span&gt;).&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
     <property name="inputMask">
      <string>HHHH</string>
     </property>
     <property name="placeholderText">
      <string>Input Item ID</string>
     </property>
    </widget>
   </item>
   <item row="1" column="3">
    <widget class="QDialogButtonBox" name="BB_PageSearchControl">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="toolTip">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Use this to navigate the page below&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Ignore|QDialogButtonBox::Retry</set>
     </property>
    </widget>
   </item>
   <item row="0" column="1">
    <widget class="QLineEdit" name="LE_PresentID">
     <property name="toolTip">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Present IDs are used to ensure that a player can't get the present more than once. Therefore, try to avoid using repeated numbers. &lt;/p&gt;&lt;p&gt;It is recommended to start at 0x100 and above, as to avoid conflicting with the Present IDs Nintendo have already used for official presents.&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600; text-decoration: underline;&quot;&gt;Note:&lt;/span&gt; ID FFFF is not a correct ID, as the game uses this for 'no ID'. It will be automatically replaced with FFFE.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
     <property name="inputMask">
      <string>HHHH</string>
     </property>
     <property name="placeholderText">
      <string>Input Present ID</string>
     </property>
    </widget>
   </item>
   <item row="2" column="0" colspan="4">
    <widget class="QTextBrowser" name="textBrowser">
     <property name="acceptDrops">
      <bool>false</bool>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Sunken</enum>
     </property>
     <property name="openLinks">
      <bool>false</bool>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>PresentCreator</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>PresentCreator</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
